### PR TITLE
Add rpc authentication support

### PR DESCRIPTION
**Pull request by jnordberg**

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -154,6 +157,23 @@ class Steem extends EventEmitter {
             .then(res => { callback(null, res) }, err => { callback(err) });
     }
 
+    signedCall(method, params, account, key, callback) {
```
Comment (goldibex) [inline src/api/index.js]: make this `async` and return the promise from `jsonRpc`

Comment (jnordberg) [reply src/api/index.js]: I thought the api methods had some thing iterating over them converting callbacks to promises

Comment (bonustrack) [reply src/api/index.js]: Currently only the steemd methods are promisified on not every methods of `steem.api` https://github.com/steemit/steem-js/blob/11ecd391a971133fca657f55aafb6e26bf4bbe2a/src/api/index.js#L49-L50

Comment (goldibex) [reply src/api/index.js]: yeah, make sure this method talks promises

**Comment by bonustrack** [comment]
The methods from `steem.api` fail for me with this error: `ReferenceError: regeneratorRuntime is not defined`. @jnordberg can you reproduce this error running the file `/examples/server.js` or `/examples/index.html`?

**Comment by jnordberg** [comment]
Bet is the introduction of `async` functions, @goldibex I'm going to revert to my initial design

**Comment by jnordberg** [comment]
@bonustrack try now

**Comment by bonustrack** [comment]
Now it's working for me @jnordberg

**Pull request merged**